.gitpod.yml

```diff
@@ -0,0 +1,10 @@
+image:
+  file: docker/dev-gitpod.Dockerfile
+
+tasks:
+  - init: |
+      pip install -e .
+      pip install -r requirements_test.txt
+      pip install -r requirements_docs.txt
+      pip install pre-commit
+      pre-commit install --install-hooks
```

Comment:
This is slow at first load, but will speed up when user does `git commit`.  I think we should remove `--install-hooks` to speed up time to first terminal prompt, but wanted feedback on what is preferred.

Reply:
Why don't we put all of this in the `Dockerfile`?

Reply:
`pip install -e .` is the only tricky one. You'll want to copy the `setup.py` ONLY to the Dockerfile then do `pip install -e .`

Reply:
> `pip install -e .` is the only tricky one. You'll want to copy the `setup.py` ONLY to the Dockerfile then do `pip install -e .`

I have to digest this comment a bit.  Do you mean copy only `setup.py` to the dockerfile and then run `pip install -e .` inside the init for gitpod?

Reply:
I'd have the Docker file looking something like:

```Dockerfile
...
RUN pip install -r requirements_test.txt
RUN pip install -r requirements_docs.txt
RUN pip install pre-commit
COPY setup.py ./setup.py
RUN pip install -e .
...
```